feature-account-impl/src/main/java/io/novafoundation/nova/feature_account_impl/presentation/mnemonic/backup/BackupMnemonicViewModel.kt

```diff
@@ -52,22 +59,18 @@ class BackupMnemonicViewModel(
     private val _showMnemonicWarningDialog = MutableLiveData<Event<Unit>>()
     val showMnemonicWarningDialog: LiveData<Event<Unit>> = _showMnemonicWarningDialog
 
-    private val warningAcceptedFlow = MutableStateFlow(false)
-
-    val mnemonicDisplay = combine(
-        mnemonicFlow,
-        warningAcceptedFlow
-    ) { mnemonic, warningAccepted ->
-        mnemonic.spacedWords().takeIf { warningAccepted }
+    val mnemonicDisplay = mnemonicFlow.map { mnemonic ->
+        mnemonic.wordList.mapIndexed { index, word ->
+            MnemonicWord(id = index, content = word, indexDisplay = index.plus(1).format(), removed = false)
+        }
     }
 
-    val continueText = flowOf {
-        val stringRes = when (payload) {
-            is BackupMnemonicPayload.Confirm -> R.string.account_confirm_mnemonic
-            is BackupMnemonicPayload.Create -> R.string.common_continue
+    val continueButtonState = combine(mnemonicWasShown, conditionsState) { mnemonicShown, conditions ->
+        val allConditionsSelected = conditions.values.size == 3 && conditions.values.all { it }
+        when {
+            mnemonicShown && allConditionsSelected -> DescriptiveButtonState.Enabled(resourceManager.getString(R.string.common_confirm))
+            else -> DescriptiveButtonState.Disabled(resourceManager.getString(R.string.backup_mnemonic_disabled_button))
```

Comment:
Looks like we show "Agree with conditions" even when all conditions are checked but mnemonic is not shown. Not sure if its expected or not